L1Trigger/L1TMuonEndCap/src/PrimitiveSelection.cc

```diff
@@ -53,48 +56,30 @@ void PrimitiveSelection::process(
     bool patchPattern = true;
-    if (patchPattern) {
-      if (new_tp.getCSCData().pattern == 11 || new_tp.getCSCData().pattern == 12) {  // 11, 12 -> 10
-	edm::LogWarning("L1T") << "\nEMTF emulator patching corrupt CSC LCT pattern: changing " << new_tp.getCSCData().pattern << " to 10\n";
+    if (patchPattern && new_tp.subsystem() == TriggerPrimitive::kCSC) {
+      if (new_tp.getCSCData().pattern == 11 || new_tp.getCSCData().pattern == 12 || new_tp.getCSCData().pattern == 13 || new_tp.getCSCData().pattern == 14) {  // 11, 12, 13, 14 -> 10
+        edm::LogWarning("L1T") << "EMTF patching corrupt CSC LCT pattern: changing " << new_tp.getCSCData().pattern << " to 10";
         new_tp.accessCSCData().pattern = 10;
       }
     }
 
     // Patch the LCT quality number
     // It should be 1-15, see: L1Trigger/CSCTriggerPrimitives/src/CSCMotherboard.cc
     bool patchQuality = true;
-    if (patchQuality) {
-      if (new_tp.subsystem() == TriggerPrimitive::kCSC && new_tp.getCSCData().quality == 0) {  // 0 -> 1
-	edm::LogWarning("L1T") << "\nEMTF emulator patching corrupt CSC LCT quality: changing " << new_tp.getCSCData().quality << " to 1\n";
+    if (patchQuality && new_tp.subsystem() == TriggerPrimitive::kCSC) {
+      if (new_tp.getCSCData().quality == 0) {  // 0 -> 1
+        edm::LogWarning("L1T") << "EMTF patching corrupt CSC LCT quality: changing " << new_tp.getCSCData().quality << " to 1";
         new_tp.accessCSCData().quality = 1;
       }
     }
 
     int selected_csc = select_csc(new_tp); // Returns CSC "link" index (0 - 53)
 
     if (selected_csc >= 0) {
-      if (not(selected_csc < NUM_CSC_CHAMBERS))
-	{ edm::LogError("L1T") << "selected_csc = " << selected_csc << ", NUM_CSC_CHAMBERS = " << NUM_CSC_CHAMBERS; return; }
-      
-      if (selected_csc_map[selected_csc].size() < 2) {
-	selected_csc_map[selected_csc].push_back(new_tp);
-      }
-      else {
-	edm::LogWarning("L1T") << "\n******************* EMTF EMULATOR: SUPER-BIZZARE CASE *******************";
-	edm::LogWarning("L1T") << "Found 3 CSC trigger primitives in the same chamber";
-	for (int ii = 0; ii < 3; ii++) {
-	  TriggerPrimitive tp_err = (ii < 2 ? selected_csc_map[selected_csc].at(ii) : new_tp);
-	  edm::LogWarning("L1T") << "LCT #" << ii+1 << ": BX " << tp_err.getBX() 
-		    << ", endcap " << tp_err.detId<CSCDetId>().endcap() << ", sector " << tp_err.detId<CSCDetId>().triggerSector()
-		    << ", station " << tp_err.detId<CSCDetId>().station() << ", ring " << tp_err.detId<CSCDetId>().ring()
-		    << ", chamber " << tp_err.detId<CSCDetId>().chamber() << ", CSC ID " << tp_err.getCSCData().cscID
-		    << ": strip " << tp_err.getStrip() << ", wire " << tp_err.getWire();
-	}
-	edm::LogWarning("L1T") << "************************* ONLY KEEP FIRST TWO *************************\n\n";
-      }
-
-    } // End conditional: if (selected_csc >= 0)
-  } // End loop: for (; tp_it != tp_end; ++tp_it)
+      assert(selected_csc < NUM_CSC_CHAMBERS);
```

Comment:
Would prefer to keep the full print-out ... this is some issue in the MPCs or the EMTF unpacker which we should try to figure out sometime.

Reply:
Wait, this is for the unpacker? I thought this was for an old emulator issue that has been fixed? Ok, I can keep it.